stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -0,0 +1,241 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .api_client import APIClient
+import json
+
+
+class InvalidMetadataException(Exception):
+    pass
+
+
+class Connector(BaseJsonSyncConnector):
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+        self.expired_token_count = 0
+
+    async def create_results_connection(self, query, offset, length, metadata=None):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :param metadata: dict
+        :return: return_obj, dict
+        """
+        data = []
+        try:
+            offset = int(offset)
+            length = int(length)
+            # result_count is overall record count from all the previous batches processed.
```

Comment:
This seems off. Why is offset being used to hold the total record count? We discussed before that there is a way to do this without worrying about the total number of results we've received.

I understand that what you are trying to handle is the remaining data from a partially handled page. IE
length = 80 page_size = 100.
The results will have a remainder of 20. 

The easiest way to do this is to simply persist how much of the page is processed as META_DATA. For example, in this above scenario, you return back
{"page":1, "page_index":80}. 

Then on the next call you return back
{"page":2, "page_index":60}

Then
{"page":3, "page_index":40}

You are keeping track of both the current page, and where you were at in that page with the metadata. The first query should handle the offset and subsequent ones should just use the page + page_index.

I don't question that this implementation doesn't work, but I am concerned that it's relying a bit too much on the caller behaving a certain way. Relying on a simple page + page_index implementation is straight-forward and easier to understand. After the first call, we just ignore offset and return length number of records.

Reply:
Updated the pagination as per the suggestion.